src/gateway/bridge/shard_queuer.rs

```diff
@@ -301,46 +301,44 @@ impl ShardQueuer {
 /// [`max_concurrency`](crate::model::gateway::SessionStartLimit::max_concurrency).
 #[must_use]
 pub struct ShardQueue {
-    buckets: HashMap<u16, VecDeque<ShardId>>,
-    max_concurrency: NonZeroU16,
+    buckets: FixedArray<VecDeque<ShardId>, u16>,
 }
 
 impl ShardQueue {
     pub fn new(max_concurrency: NonZeroU16) -> Self {
+        let buckets = vec![VecDeque::new(); max_concurrency.get() as usize].into_boxed_slice();
+        let buckets = FixedArray::try_from(buckets).expect("should fit without truncation");
+
         Self {
-            buckets: HashMap::with_capacity(max_concurrency.get() as usize),
-            max_concurrency,
+            buckets,
         }
     }
 
+    fn calculate_bucket(&self, shard_id: ShardId) -> usize {
+        let max_concurrency = u16::try_from(self.buckets.len()).expect("length is stored as u16");
+        usize::from(shard_id.0 % max_concurrency)
```

Comment:
```suggestion
        usize::from(shard_id.0) % self.buckets.len()
```
We don't need to be so verbose.

Reply:
This doesn't work, as buckets.len returns u32 because the max length type of a FixedArray is u32... although this is kind of weird and I'm going to go fix that for small-fixed-array 0.3

Reply:
Done.